### DISCUSSION TITLE
How to Submit a Package Update?

### DISCUSSION THREAD

**Discussion by teklot** [microsoft/vcpkg#31443] [Q&A] [answered]
Which files needed to be changed for a package update? In other words, what is the process of a pull request for a package update?

**Comment by AugP** [comment]
We have a few documentation articles on how to create or update ports on the Microsoft Learn site. One example is here: https://learn.microsoft.com/en-us/vcpkg/examples/patching. Also see the table of contents on the left with more links. Let me know if you are looking for something more specific.

**Reply by teklot** [reply]
Thanks for the info. In my case the port (quill) has already been updated and released by the original developer. It seems updating the "ports/quill/vcpkg.json" (bump the version from 2.8.0 to 2.9.0 in this case) and creating a pull request afterwards would do the job. Can you confirm that please?

**Reply by Neumann-A** [reply, marked as answer]
> It seems updating the "ports/quill/vcpkg.json"

That is one port you also need to update the `SHA` in "ports/quill/portfile.cmake" and maybe also adjust the `REF` so it actually downloads 2.9.0. After that you have to update the version database as pointed out by the PR checklist.

**Reply by teklot** [reply]
Great! Where do I get the updated `SHA` from?

**Reply by Neumann-A** [reply]
There is more than one way to obtain it. What i typically do is set it to 0 and run vcpkg with it. It will error out and tell you the actual SHA in the error message.

**Reply by teklot** [reply]
It worked! Thanks everyone.

**Reply by teklot** [reply]
How about the git-tree hash for the version database? In my case it is located in "versions/q-/quill.json" but cannot figure out where the git-tree hashes came from.

**Reply by Neumann-A** [reply]
`vcpkg x-add-version <portname or --all>` but that should be listed as a point in the PR checklist:

>  The version database is fixed by rerunning ./vcpkg x-add-version --all and committing the result.

**Reply by teklot** [reply]
Worked like a charm. Thanks!
